core/server/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -155,10 +160,27 @@ public synchronized AlluxioURI resolve(AlluxioURI uri) throws InvalidPathExcepti
     LOG.debug("Resolving {}", path);
     String mountPoint = getMountPoint(uri);
     if (mountPoint != null) {
-      AlluxioURI ufsPath = mMountTable.get(mountPoint);
+      AlluxioURI ufsPath = mMountTable.get(mountPoint).getUfsUri();
       return new AlluxioURI(ufsPath.getScheme(), ufsPath.getAuthority(),
           PathUtils.concatPath(ufsPath.getPath(), path.substring(mountPoint.length())));
     }
     return uri;
   }
+
+  /**
+   * Checks to see if a write operation is allowed for the specified Alluxio path, by determining
+   * if it is under a readonly mount point.
+   *
+   * @param alluxioUri an Alluxio path URI
+   * @throws InvalidPathException if the Alluxio path is invalid
+   * @throws AccessControlException if the Alluxio path is under a readonly mount point
+   */
+  public synchronized void checkWritable(AlluxioURI alluxioUri)
```

Comment:
Are you saying this should return a boolean, and not throw an exception? I didn't want to fold this into access control, since that could be turned off and on via a flag, and a readonly mount should always be readonly.